src/generator/operations.ts

```diff
@@ -1078,34 +1080,15 @@ function generateARMLROBeginMethod(op: Operation, imports: ImportManager): strin
   if (op.extensions?.['x-ms-long-running-operation-options']?.['final-state-via']) {
     finalState = op.extensions?.['x-ms-long-running-operation-options']?.['final-state-via'];
   }
-  text += `\tpt, err := armcore.NewPoller("${clientName}.${op.language.go!.name}", "${finalState}", resp, client.${info.protocolNaming.errorMethod})\n`;
+  if (isARM) {
+    text += `\tpt, err := armcore.NewPoller("${clientName}.${op.language.go!.name}", "${finalState}", resp, client.${info.protocolNaming.errorMethod})\n`;
+  } else {
+    text += `\tpt, err := azcore.NewLROPoller("${clientName}.${op.language.go!.name}",resp, client.con.Pipeline(), client.${info.protocolNaming.errorMethod})\n`;
+  }
   text += '\tif err != nil {\n';
   text += `\t\treturn ${zeroResp}, err\n`;
   text += '\t}\n';
-  text += `\tpoller := &${internalPollerTypeName(<PollerInfo>op.language.go!.pollerType)}{\n`;
-  text += '\t\tpt: pt,\n';
-  if (isPageableOperation(op)) {
-    const statusCodes = getStatusCodes(op);
-    if (statusCodes.indexOf('200') < 0) {
-      statusCodes.push('200');
-    }
-    if (statusCodes.indexOf('204') < 0) {
-      statusCodes.push('204');
-    }
-    statusCodes.sort();
-    text += `\t\terrHandler: func(resp *azcore.Response) error {\n`;
-    text += `\t\t\tif resp.HasStatusCode(${formatStatusCodes(statusCodes)}) {\n`;
-    text += `\t\t\t\treturn nil\n`;
-    text += '\t\t\t}\n';
-    text += `\t\t\treturn client.${info.protocolNaming.errorMethod}(resp)\n`;
-    text += '\t\t},\n';
-    text += `\t\trespHandler: func(resp *azcore.Response) (${schemaResponse!.schema.language.go!.responseType.name}, error) {\n`;
-    text += generateResponseUnmarshaller(op, schemaResponse!, imports);
-    text += '\t\t},\n';
-    text += `\t\tstatusCodes: []int{${formatStatusCodes(statusCodes)}},\n`;
-  }
-  text += '\t\tpipeline: client.con.Pipeline(),\n';
-  text += '\t}\n';
+  text += emitPoller(op, isARM, imports);
```

Comment:
A true nit 😆 but I would switch the param order to end with `isARM` to match the other function signatures.

Reply:
Fixed